### PR TITLE
Make slug editable along with validation

### PR DESCRIPTION
**Pull request by AmitJoki**
Before:

image

After:

image

### CODE REVIEW

File: app/models/campaign.rb [hunk 1]
```diff
@@ -156,8 +156,9 @@ def valid_start_and_end_dates?
   end
 
   def set_slug
+    campaign_slug = slug.present? ? slug : title
     # Strip everything but unicode letters and digits, and convert spaces to underscores.
-    self.slug = title.downcase.gsub(/[^\p{L}0-9 ]/, '').tr(' ', '_') unless slug.present?
+    self.slug = campaign_slug.downcase.gsub(/[^\p{L}0-9 ]/, '').tr(' ', '_')
```
Comment (ragesoss) [inline app/models/campaign.rb]: Looks like this change is causing some test failures, probably in connection with the default campaign in the fixture.

Comment (AmitJoki) [reply app/models/campaign.rb]: It is passing now. Should only normalize if it is not already valid.

**Comment by bwreid** [comment]
@AmitJoki This looks good to me. We'll probably wait to officially merge this until Sage returns, but I don't believe any further work is required. Nice job!

**Comment by AmitJoki** [comment]
@ragesoss can we get this merged?

**Pull request merged**